app/build.gradle

```diff
@@ -28,7 +29,15 @@ android {
         debug {
             multiDexEnabled true
             debuggable true
-            applicationIdSuffix ".debug"
+
+            def workingBranch = "git rev-parse --abbrev-ref HEAD".execute().text.trim()
+            if (workingBranch.isEmpty() || workingBranch == "master" || workingBranch == "dev") {
+                applicationIdSuffix ".debug"
+                resValue "string", "app_name", "NewPipe Debug"
```

Comment:
Uh, won't it add ``.debug`` and ``" Debug" to master? ````|| workingBranch == "master"``

Have you tried to apply your changes in the master branch and launch the app, to see what happens?

Reply:
The branch name is independent from the build type. So we can have a release build on any branch and a debug build on the master branch. Gradle has no idea whether the current branch is a development branch or not, so it can't do decisions based on that.
So it is ok for a debug build on master branch to have .debug at the end